File: src/main/java/org/location/model/Rent.java
package org.location.model;

import lombok.Getter;

import java.time.LocalDate;

@Getter
public class Rent {
    private final LocalDate startDate;
    private final LocalDate endDate;
    private final int days;
    private final double price;
    private final ItemForRent itemForRent;

    public Rent(LocalDate startDate, int days, ItemForRent itemForRent) {
        if (days < 1) {
            throw new IllegalArgumentException("days must be at least one");
        }
        this.startDate = startDate;
        this.days = days;
        this.endDate = startDate.plusDays(days);
        this.itemForRent = itemForRent;
        this.price = itemForRent.getPricePerDay() * this.days;
    }

    @Override
    public String toString() {
        return String.format("Rent from %s to %s (%d days) - $%.2f [%s]",
                startDate, endDate, days, price, itemForRent.getName());
    }
}

File: Rent.js
class Rent {
    constructor(startDate, days, itemForRent) {
      if (days < 1) throw new Error("days must be at least one");
      this.startDate = new Date(startDate);
      this.days = days;
      this.endDate = new Date(this.startDate);
      this.endDate.setDate(this.startDate.getDate() + days);
      this.itemForRent = itemForRent;
      this.price = itemForRent.pricePerDay * days;
    }
  
    toString() {
      return `Rent from ${this.startDate.toISOString().split('T')[0]} to ${this.endDate.toISOString().split('T')[0]} (${this.days} days) - $${this.price.toFixed(2)} [${this.itemForRent.name}]`;
    }
  }
  
  module.exports = Rent;
  